File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/playerstate/PersonalTimeSetting.java
package top.likoslupus.cellulosesz.api.playerstate;

public sealed interface PersonalTimeSetting permits
        PersonalTimeSetting.Fixed,
        PersonalTimeSetting.Relative,
        PersonalTimeSetting.Reset {

    static PersonalTimeSetting reset() {
        return new Reset();
    }

    record Fixed(
            long ticks
    ) implements PersonalTimeSetting {

    }

    record Relative(
            long offset
    ) implements PersonalTimeSetting {

    }

    record Reset() implements PersonalTimeSetting {

    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/playerstate/PersonalTimeSetting.kt
package top.likoslupus.cellulosesz.api.playerstate

public sealed interface PersonalTimeSetting {

    @JvmRecord
    public data class Fixed(
        public val ticks: Long
    ) : PersonalTimeSetting

    @JvmRecord
    public data class Relative(
        public val offset: Long
    ) : PersonalTimeSetting

    public data object Reset : PersonalTimeSetting

    public companion object {

        @JvmStatic
        public fun reset(): PersonalTimeSetting = Reset

    }

}
